usr/lib/common/new_host.c

```diff
@@ -115,13 +115,27 @@ CK_RV ST_Initialize(API_Slot_t *sltp, CK_SLOT_ID SlotNumber,
     bt_init(&sltp->TokData->publ_token_obj_btree, call_object_free);
 
     if (strlen(sinfp->tokname)) {
-        sprintf(abs_tokdir_name, "%s/%s", CONFIG_PATH, sinfp->tokname);
+        int snres;
```

Comment:
Not sure if it is a good idea to declare a variable in an if block. I remember that I  run into problems with such for certain distros, where the compiler flagged this as an error... Better declare the variable at the beginning of the function

Reply:
Ingo this is C standart. With every block you can declare variables. What's the issue with this ?
I know that C99 and C++ allow variables in the middle of code, but Ansi C90 restricts to the beginnning
of a code block - but not to the beginning of a function.

Reply:
Maybe I confused this with another case. Back then I hade something like this:
`for(int i=0, .... ` and the compiler flagged this as only allowed in C99.

Reply:
I moved it to the top of the function.

Reply:
and ingo the compiler is right. A variable declaration like you show here is not allowed with Ansi C.

Reply:
I like to declare a variable that is only used within a block at the top of that block to limit its scope. Such code is usually easier to review (and conforms to ansi c). Declaring variables in the middle of a code block or for loop is definitely not ansi c and IMO should be avoided.

Reply:
However, im also fine with declaring all variables at the top of a function. Variables that are intended to be used only inside the scope of a block statement should then only be initialized inside that block such that you get an undef var warning when using the variable out of its intended scope. Just my 2 cents..